### ISSUE TITLE
BERT does mask-answering or sequence prediction or both???

### ISSUE THREAD

**Issue by bladedsupernova** [huggingface/pytorch-pretrained-BERT#481]
I'm working on a exciting project but need to know something fast.

I seen BERT adds a word at the end of your input, like sequence prediction, elongating your input text.

But I read that BERT has been trained at (and is a pro at) filling in the blank word mask, and in fact CANNOT do sequence prediction at the end of one's input (contradiction!!). When BERT adds a fill-in answer where the mask sits, it goes in context.

I want a tool that returns me a similar word in context (the mask ANSWER!!), but the colab tool I tried is not a mask/answer giver, ..it adds words at the endddd of my input lol!!!

**Comment by bladedsupernova** [comment]
When BERT fills-in a MASK, does it always use a answer it has seen or does it generalize and create never-seen Q-As? Ex. "The *dog* in the cat, by Dr. Souse the Cat"
